web_client/dialogs/editRegionOfInterest.js

```diff
@@ -0,0 +1,194 @@
+    /**
+     * Convert from magnification to zoom level.
+     */
+    magnificationToZoom(magnification) {
+        return parseFloat(this.areaElement.maxZoom) -
+            Math.log2(this.areaElement.maxMag / magnification);
+    },
+
+    /**
+     * Convert from magnification to zoom level.
+     */
+    scaleBounds() {
+        var zoom = this.magnificationToZoom(this._magnification);
+        var factor = Math.pow(2, zoom - this.areaElement.maxZoom);
+        var scaleWidth = Math.round(factor * this.areaElement.width);
+        var scaleHeight = Math.round(factor * this.areaElement.height);
+        return { 'width': scaleWidth, 'height': scaleHeight };
+    },
+
+    /**
+     * Get the number of pixel in the region of interest
+     */
+    getNumberPixels() {
+        var bounds = this.scaleBounds();
+        var pixelNumber = bounds.width * bounds.height;
+        return pixelNumber;
+    },
+
+    /**
+     * Get the size of the file before download it for an image in 24b/px (result in Bytes)
+     */
+    getFileSize() {
+        var fileSize = (this.getNumberPixels() * 3 + this._sizeCte) * this._compressionRatio;
+        return fileSize;
+    },
+
+    /**
+     * Get the size of the file in the appropriate unity (Bytes, MB, GB...)
+     */
+    getConvertFileSize() {
+        var bytesNumber = this.getFileSize();
+        var convertedSize = formatSize(bytesNumber);
+        this.downloadDisable(bytesNumber >= Math.pow(2, 30));
+        return convertedSize;
+    },
+
+    /**
+     * Disable the Download button if SizeFile > 1GB
+     */
+    downloadDisable(bool) {
+        if (bool) {
+            this.$('#h-download-area-link').attr('disabled', 'disabled');
+            this.$('#h-msgDisable').removeClass('hidden');
+            this.$('#h-download-area-link').bind('click', (ev) => ev.preventDefault());
+        } else {
+            this.$('#h-download-area-link').removeAttr('disabled').unbind('click');
+            this.$('#h-msgDisable').addClass('hidden');
+        }
+    },
+
+    /**
+     * Set the size of the file, bounds, format...
+     * And download the image
+     */
+    updateform(evt) {
+        // Find the good compresion ration there are random now
+        const selectedOption = $('#h-download-image-format option:selected').text();
+        const bounds = this.scaleBounds();
+        switch (selectedOption) {
+            case 'JPEG':
+                this._format = 'JPEG';
+                this._compressionRatio = 0.35;
+                break;
+            case 'PNG':
+                this._format = 'PNG';
+                this._compressionRatio = 0.7;
+                break;
+            case 'TIFF':
+                this._format = 'TIFF';
+                this._compressionRatio = 0.8;
+                break;
+            default:
+                // JPEG is the default format
+                this._compressionRatio = 0.35;
+        }
+        this._magnification = parseFloat($('#h-element-mag').val());
```

Comment:
The `const bounds = this.scaleBounds();` line above needs to come after the `_magnification` is set, otherwise the bounds  aren't correct.